File: src (cpp)/animal/mammalia/tiger.cpp
#include "tiger.h"

Tiger::Tiger(int _x, int _y, int _weight) : def_ID(3),
                                            def_ratio_meat(50),
                                            def_ratio_plant(0) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Tiger::Interact() {
  std::cout << "The tiger is graciously lying on the grass" << std::endl;
}

File: src/virtualzoo/animal/mammalia/Tiger.java
package virtualzoo.animal.mammalia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas Tiger mendefinisikan atribut untuk tiger.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Tiger extends Mammalia {

  /**
   * Constructor
   *
   * Menciptakan tiger dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat tiger.
   */
  public Tiger(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Carnivora(animalWeight, 50);
    ID = 3;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan tiger.
   */
  public void interact() {
    System.out.println("The tiger is graciously lying on the grass");
  }
}
